src/app/static/src/app/apiService.ts

```diff
@@ -49,8 +51,18 @@ export class APIService<S extends string, E extends string> implements API<S, E>
     private _freezeResponse: Boolean = false;
 
     static getFirstErrorFromFailure = (failure: Response) => {
-        const firstError = failure.errors[0];
-        return firstError.detail ? firstError.detail : firstError.error;
+        const firstError = failure.errors && failure.errors[0];
+        let result;
+        if (firstError) {
+            result =  firstError.detail ? firstError.detail : firstError.error;
+        }
+
+        if (!result)
```

Comment:
The idea is that we have already checked that the API response is correctly formatted before calling this method, and thrown an error saying "Could not parse API response" if not.  So no need to do all these null checks here. That check happens [here](https://github.com/mrc-ide/hint/pull/174/files#diff-0f0781cfcf5e78654409fce90c2c466fR116). Could make it more complete though I agree, by checking that `errors` exists and is non-empty/correctly formatted.

Reply:
Yeah, I put that in because errors was undefined in my test case (which I think I'd forced by killing the hintr) container. Good idea to move that check to the format checking part.